src/objects.c

```diff
@@ -1799,3 +1813,748 @@ int p11prov_obj_export_public_ec_key(P11PROV_OBJ *obj, OSSL_CALLBACK *cb_fn,
+            goto done;
+        }
+    }
+    tmp.key = "EC Group";
+    tmp.data = &curve_nid;
+    tmp.data_size = sizeof(curve_nid);
+    rv = param_to_attr(ctx, &tmp, tmp.key, &findctx->attrs[findctx->numattrs],
+                       CKA_P11PROV_CURVE_NID, false);
+    if (rv != CKR_OK) {
+        goto done;
+    }
+    findctx->numattrs++;
+
+    if (curve_name != NULL) {
+        tmp.key = "EC Curve Name";
+        tmp.data = (void *)curve_name;
+        tmp.data_size = strlen(curve_name) + 1;
+        rv = param_to_attr(ctx, &tmp, tmp.key,
+                           &findctx->attrs[findctx->numattrs],
+                           CKA_P11PROV_CURVE_NAME, false);
+        if (rv != CKR_OK) {
+            goto done;
+        }
+        findctx->numattrs++;
+    }
+
+    findctx->bit_size = EC_GROUP_order_bits(group);
+    findctx->key_size = (findctx->bit_size + 7) / 8;
+    rv = CKR_OK;
+
+done:
+    EC_GROUP_free(group);
+    return rv;
+}
+
+static CK_RV return_dup_key(P11PROV_OBJ *dst, P11PROV_OBJ *src)
+{
+    CK_RV rv;
+
+    dst->slotid = src->slotid;
+    dst->handle = src->handle;
+    dst->class = src->class;
+    dst->cka_copyable = src->cka_copyable;
+    dst->cka_token = src->cka_token;
+    dst->data.key = src->data.key;
+
+    dst->attrs = OPENSSL_malloc(sizeof(CK_ATTRIBUTE) * src->numattrs);
+    if (!dst->attrs) {
+        rv = CKR_HOST_MEMORY;
+        P11PROV_raise(dst->ctx, rv, "Failed allocation");
+        return rv;
+        ;
+    }
+    dst->numattrs = 0;
+    for (int i = 0; i < src->numattrs; i++) {
+        rv = p11prov_copy_attr(&dst->attrs[i], &src->attrs[i]);
+        if (rv != CKR_OK) {
+            rv = CKR_HOST_MEMORY;
+            P11PROV_raise(dst->ctx, rv, "Failed attr copy");
+            return rv;
+        }
+        dst->numattrs++;
+    }
+
+    return CKR_OK;
+}
+
+static CK_RV fix_ec_key_import(P11PROV_OBJ *key, int allocattrs)
+{
+    CK_ATTRIBUTE *pub;
+    ASN1_OCTET_STRING oct;
+    unsigned char *der = NULL;
+    int len;
+
+    if (key->numattrs >= allocattrs) {
+        P11PROV_raise(key->ctx, CKR_GENERAL_ERROR,
+                      "Too many attributes?? %d >= %d", key->numattrs,
+                      allocattrs);
+        return CKR_GENERAL_ERROR;
+    }
+
+    pub = p11prov_obj_get_attr(key, CKA_P11PROV_PUB_KEY);
+    if (!pub) {
+        P11PROV_raise(key->ctx, CKR_KEY_INDIGESTIBLE, "No public key found");
+        return CKR_KEY_INDIGESTIBLE;
+    }
+
+    oct.data = pub->pValue;
+    oct.length = pub->ulValueLen;
+    oct.flags = 0;
+
+    len = i2d_ASN1_OCTET_STRING(&oct, &der);
+    if (len < 0) {
+        P11PROV_raise(key->ctx, CKR_KEY_INDIGESTIBLE,
+                      "Failure to encode EC point to DER");
+        return CKR_KEY_INDIGESTIBLE;
+    }
+    key->attrs[key->numattrs].type = CKA_EC_POINT;
+    key->attrs[key->numattrs].pValue = der;
+    key->attrs[key->numattrs].ulValueLen = len;
+    key->numattrs++;
+
+    /* TODO: EC_PARAMS */
```

Comment:
is this todo still valid? If so, I think it needs some more description as it is not clear to me what needs to be done and why.